tests/web/test_api_v1.py

```diff
@@ -76,3 +76,46 @@ def test_request_logs_contain_required_info(mocker):
     assert extra_dict["path"] == root_path
     assert extra_dict["session_id"] == sid
     assert extra_dict["sequence_no"] == seq
+
+
+# The first two IP addresses are taken from `GeoLite2-City-Test.mmdb`
+@pytest.mark.parametrize(
+    "ip,expected_country,expected_region,expected_city,expected_dma",
+    [
+        ("216.160.83.56", "US", "WA", "Milton", 819),
+        ("2.125.160.216", "GB", "ENG", "Boxford", None),
+        ("127.0.0.1", None, None, None, None),
+    ],
+)
+def test_geolocation(
+    mocker, ip, expected_country, expected_region, expected_city, expected_dma
+):
+    import logging
+
+    mock_client = mocker.patch("fastapi.Request.client")
+    mock_client.host = ip
+
+    spy = mocker.spy(logging.Logger, "info")
+    query = "nope"
+    root_path = "/api/v1/suggest"
+    client.get(f"{root_path}?q={query}")
```

Comment:
Should be consistent with the query string in `test_geolocation_with_invalid_ip`?
```suggestion
    client.get("/api/v1/suggest?q=nope")
```